streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -488,7 +493,28 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
           registerReceiver(streamId, typ, host, executorId, receiverEndpoint, context.senderAddress)
         context.reply(successful)
       case AddBlock(receivedBlockInfo) =>
-        context.reply(addBlock(receivedBlockInfo))
+        if (WriteAheadLogUtils.isBatchingEnabled(ssc.conf, isDriver = true)) {
+          val f = Future {
+            synchronized {
+              if (active) {
```

Comment:
You are calling the whole `addBlock` in `synchronized`. Isnt that what we wanted to avoid originally, so that the method does not block other threads why the WAL is being written (that why we removed synchronized inside the whole method of addBlock and made it very specific on a few lines only)?